### PR TITLE
feat(forms): add Field.Indeterminate

### PR DESCRIPTION
**Pull request by tujoworker**

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Indeterminate/info.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { Field } from '@dnb/eufemia/extensions/forms'
+render(
+  <Field.Indeterminate
+    dependencePaths={['/checkbox1', '/checkbox2', '/checkbox3']}
+    path="/checkboxParent"
+  />,
+)
+```
+
+It should only be used in combination with checkbox looking variant.
```
Comment (joakbjerk) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Indeterminate/info.mdx:19]:
```suggestion
It should only be used in combination with checkbox looking variants.
```

Comment (joakbjerk) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Indeterminate/info.mdx:19]: Forgot to add, looks very good 😄

**Comment by AadneRo** [comment]
Is this the functionality we want? I would expect the parent checkbox to uncheck all the children also 


https://github.com/dnbexperience/eufemia/assets/41082005/c5dc60bc-8671-4f6d-babf-9d0db421b51c

File: packages/dnb-eufemia/src/components/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -202,7 +202,7 @@ function Checkbox(localProps: CheckboxProps) {
 
   useEffect(() => {
     ref.current.indeterminate = indeterminate
-  }, [indeterminate])
+  }, [indeterminate, ref])
```
Comment (AadneRo) [inline packages/dnb-eufemia/src/components/checkbox/Checkbox.tsx:205]:
```suggestion
  }, [indeterminate])
```
https://github.com/dnbexperience/eufemia/pull/3515#discussion_r1593769308

**Comment by tujoworker** [comment]
Yeah, it should uncheck/toggle.

**Review by AadneRo — APPROVED** [review]

**Pull request merged**